coral/src/app/components/Form.tsx

```diff
@@ -145,6 +145,44 @@ export const TextInput = <T extends FieldValues>(
 
 TextInput.Skeleton = BaseInput.Skeleton;
 
+//
+// <NumberInput>
+//
+function _NumberInput<T extends FieldValues>({
+  name,
+  formContext: form,
+  ...props
+}: BaseInputProps & FormInputProps<T> & FormRegisterProps<T>) {
+  const { errors } = form.formState;
+  const error = get(errors, name)?.message as string;
```

Comment:
Question: this seems unsafe to my naive eyes, but then I saw this and my mind was blown. What is happening here exactly? ^^

![Screenshot 2023-01-03 at 15 47 12](https://user-images.githubusercontent.com/20607294/210380346-71963f7b-4a64-4c33-8ee0-5eafb3230deb.png)

Reply:
The implementation is from the upstream project. However, I just amended this PR to include alternative error parsing. Hopefully that is more readable due to the improved typeguards.

Reply:
Excellent!